packages/protocol/test/common/integration.ts

```diff
@@ -382,180 +382,185 @@ contract('Integration: Governance', (accounts: string[]) => {
+      // Note that this test relies on having purchased stable token in the previous test.
+      describe('When selling stable token', () => {
+        before(async () => {
+          originalStable = await stableToken.balanceOf(accounts[0])
+          originalGold = await goldToken.balanceOf(accounts[0])
+          originalReserve = await goldToken.balanceOf(reserve.address)
+          await stableToken.approve(exchange.address, sellAmount)
+          // Cannot sell more than was purchased in the previous test.
+          await exchange.sell(sellAmount.div(20), minBuyAmount, false)
+          finalStable = await stableToken.balanceOf(accounts[0])
+          finalGold = await goldToken.balanceOf(accounts[0])
+          finalReserve = await goldToken.balanceOf(reserve.address)
+        })
+
+        it(`should reduce user's stable`, async () => {
+          assert.isTrue(finalStable.lt(originalStable))
+        })
+
+        it(`should increase user's gold`, async () => {
+          assert.isTrue(finalGold.gt(originalGold))
+        })
+
+        it(`should reduce Reserve's gold`, async () => {
+          assert.isTrue(finalReserve.lt(originalReserve))
+        })
       })
     })
-  })
-
-  describe('Buying', () => {
-    const buyAmount = new BigNumber(1)
-    const maxSellAmount = new BigNumber('1000000000000000000')
-
-    describe('When buying stable token', () => {
-      before(async () => {
-        originalStable = await stableToken.balanceOf(accounts[0])
-        originalGold = await goldToken.balanceOf(accounts[0])
-        originalReserve = await goldToken.balanceOf(reserve.address)
-        await goldToken.approve(exchange.address, maxSellAmount)
-        await exchange.buy(buyAmount, maxSellAmount, false)
-        finalStable = await stableToken.balanceOf(accounts[0])
-        finalGold = await goldToken.balanceOf(accounts[0])
-        finalReserve = await goldToken.balanceOf(reserve.address)
-      })
 
-      it(`should increase user's stable`, async () => {
-        assert.isTrue(finalStable.gt(originalStable))
+    describe('Buying', () => {
+      const buyAmount = new BigNumber(100)
+      const maxSellAmount = new BigNumber('1000000000000000000')
+
+      describe('When buying stable token', () => {
+        before(async () => {
+          originalStable = await stableToken.balanceOf(accounts[0])
+          originalGold = await goldToken.balanceOf(accounts[0])
+          originalReserve = await goldToken.balanceOf(reserve.address)
+          await goldToken.approve(exchange.address, maxSellAmount)
+          await exchange.buy(buyAmount, maxSellAmount, false)
+          finalStable = await stableToken.balanceOf(accounts[0])
+          finalGold = await goldToken.balanceOf(accounts[0])
+          finalReserve = await goldToken.balanceOf(reserve.address)
+        })
+
+        it(`should increase user's stable`, async () => {
+          assert.isTrue(finalStable.gt(originalStable))
```

Comment:
Can we test that it's greater than by the expected amount? (similar for the rest of the tests)

Reply:
These tests were not written for this PR. If we want to improve the correctness we can save this for a followup PR. In the spirit of getting this release to the auditors asap, I merged to `v3.pre-audit`.